docs/topics/migrations.txt

```diff
@@ -218,6 +218,21 @@ Apps without migrations must not have relations (``ForeignKey``,
 ``ManyToManyField``, etc.) to apps with migrations. Sometimes it may work, but
 it's not supported.
 
+Swappable dependencies
+----------------------
+
+.. function:: django.db.migrations.swappable_dependency(value)
```

Comment:
```diff
 The ``swappable_dependency()`` function is used by migration operations to
-declare "swappable" dependencies such as swappable user models.
+declare "swappable" dependencies such as swappable user models. It just
+makes dependencies on the first migrations of the app instead of actual
+model name.
```
Does it look better here?

Reply:
@DevilsAutumn As far as I'm aware, Shai suggested to describe an expected `value`, e.g.

_"... declare "swappable" dependencies such as swappable user models. The argument ``value`` is a string
``"<app label>.<model>"`` describing an app label and a model name, e.g. ``` ``"myapp.MyModel"`` ```"_

Reply:
> (note also -- @felixxm I'm not sure if this belongs in this documentation, or in its own bug report -- the current code ignores the actual model name, and just makes the dependency on the first migration of the app. If, for some reason, anyone tries to use a swappable dependency with a model that is only created in the second migration of its app, things may break)

Tricky, but I would process it separately.

Reply:
@felixxm I changed that locally, but just for more clarification, i thought of adding this line too. 😄

Reply:
> > (note also -- @felixxm I'm not sure if this belongs in this documentation, or in its own bug report -- the current code ignores the actual model name, and just makes the dependency on the first migration of the app. If, for some reason, anyone tries to use a swappable dependency with a model that is only created in the second migration of its app, things may break)
> 
> Tricky, but I would process it separately.

I disagree. The currently proposed documentation is simply wrong. The key is, if the swapped-in model isn't created as part of the initial migration of an app, the dependencies can be wrong, causing migration failures.

Reply:
> The phrase `declare "swappable" dependencies such as swappable user models` isn't really correct. The `swappable_dependency()` function essentially creates a dependency on the first migration in the app of the swapped-in model. That means, it essentially adds `('auth', '__first__')` if `AUTH_USER_MODEL='auth.User'` and `('accounts', '__first__')` if `AUTH_USER_MODEL='accounts.MyUser'`.

Yeah, that was also pointed out by [Shai](https://github.com/django/django/pull/16861#discussion_r1212987567). Agreed, let's be more specific here.

Reply:
> > > (note also -- @felixxm I'm not sure if this belongs in this documentation, or in its own bug report -- the current code ignores the actual model name, and just makes the dependency on the first migration of the app. If, for some reason, anyone tries to use a swappable dependency with a model that is only created in the second migration of its app, things may break)
> > 
> > 
> > Tricky, but I would process it separately.
> 
> I disagree. The currently proposed documentation is simply wrong. The key is, if the swapped-in model isn't created as part of the initial migration of an app, the dependencies can be wrong, causing migration failures.

I took your opinion and _"my heart is still open"_ :wink: :green_heart: Let's correct this.

Reply:
![Screenshot 2023-06-02 154102](https://github.com/django/django/assets/83907321/c3c9f678-1edf-46e1-b0bb-d9845aed9a39)

Reply:
There is no need to push the `value` description to a note. Also, please push edits instead of images, they are not really reviewable.

Reply:
> > (note also -- @felixxm I'm not sure if this belongs in this documentation, or in its own bug report -- the current code ignores the actual model name, and just makes the dependency on the first migration of the app. If, for some reason, anyone tries to use a swappable dependency with a model that is only created in the second migration of its app, things may break)
> 
> Tricky, but I would process it separately.

https://code.djangoproject.com/ticket/34623